src/object/obj_verify.c

```diff
@@ -710,6 +710,11 @@ dc_obj_verify_rdg(struct dc_object *obj, struct dc_obj_verify_args *dova,
 		for (i = 1; i < reps; i++) {
 			rc = dc_obj_verify_cmp(&dova[0], &dova[i],
 					       oid, reps, start, start + i);
+			if (rc == -DER_CSUM) {
+				D_ERROR("Failed to verify because of "
+					"data corruption");
+				D_GOTO(out, rc = -DER_MISMATCH);
```

Comment:
Isn't DER_MISMATCH supposed to be for pool version mismatch?   Seems we've overloaded it here?   What does it mean for object verification?   That the object didn't match expected value ?

Reply:
We reuse the error# "DER_MISMATCH" for the case of data inconsistency between two shards of replicated object.

Reply:
@jolivier23 - are you okay that it's reused in this way? I believe if we did want to change it, it would probably fall out of the scope of this patch and might want to create a ticket for it.

Reply:
Yeah, I'm ok with keeping it as is for the purpose of the patch.   I'd prefer adding a new error though in general rather than overloading existing ones as it's confusing to see -DER_MISMATCH and then go read the documentation for that error in this case.